WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/EpilogueBuilder.kt

```diff
@@ -0,0 +1,87 @@
+package org.wordpress.android.ui.bloggingreminders
+
+import android.os.Build.VERSION_CODES
+import androidx.annotation.RequiresApi
+import org.wordpress.android.R.drawable
+import org.wordpress.android.R.string
+import org.wordpress.android.fluxc.model.BloggingRemindersModel
+import org.wordpress.android.ui.bloggingreminders.BloggingRemindersItem.Caption
+import org.wordpress.android.ui.bloggingreminders.BloggingRemindersItem.HighEmphasisText
+import org.wordpress.android.ui.bloggingreminders.BloggingRemindersItem.Illustration
+import org.wordpress.android.ui.bloggingreminders.BloggingRemindersItem.Title
+import org.wordpress.android.ui.bloggingreminders.BloggingRemindersViewModel.UiState.PrimaryButton
+import org.wordpress.android.ui.utils.ListItemInteraction
+import org.wordpress.android.ui.utils.UiString.UiStringRes
+import org.wordpress.android.ui.utils.UiString.UiStringResWithParams
+import org.wordpress.android.ui.utils.UiString.UiStringText
+import java.time.DayOfWeek
+import java.time.format.TextStyle
+import java.util.Locale
+import javax.inject.Inject
+
+class EpilogueBuilder @Inject constructor(
+    private val dayLabelUtils: DayLabelUtils
+) {
+    @RequiresApi(VERSION_CODES.O) // TODO: Remove this annotation once de-sugaring PR is merged
+    fun buildUiItems(
+        bloggingRemindersModel: BloggingRemindersModel?
+    ): List<BloggingRemindersItem> {
+        val numberOfTimes = dayLabelUtils.buildNTimesLabel(bloggingRemindersModel)
+        val enabledDays = bloggingRemindersModel?.enabledDays?.map { DayOfWeek.valueOf(it.name) }
+
+        val selectedDays = when (enabledDays?.count()) {
+            ONE_DAY -> enabledDays.joinToString { formattedDay(it) }
+            TWO_DAYS -> enabledDays.joinToString(separator = " and ") { formattedDay(it) }
+            in THREE_DAYS..SIX_DAYS -> {
+                val firstDays = enabledDays?.dropLast(1)
+                val lastDay = enabledDays?.drop(enabledDays.count() - 1)
+                firstDays?.joinToString(postfix = " and ") {
+                    formattedDay(it)
+                } + lastDay?.joinToString { formattedDay(it) }
+            }
```

Comment:
After discussing about this with the iOS folks, I believe that instead of trying to handle this logic ourselves, I'd recommend you to take a look at the [`ListFormatter`](https://developer.android.com/reference/android/icu/text/ListFormatter) class, which should make this way easier and much more accurate in terms of i18n.

Reply:
ListFormatter is only available from android O.  I was looking at using [Phrase](https://github.com/square/phrase) incase desugaring doesn't get merged or do something feasible for now